### PR TITLE
Use icon hints

### PR DESCRIPTION
**Pull request by mathstuf**
---
[udisks1 docs](http://udisks.freedesktop.org/docs/1.0.5/Device.html#Device:DevicePresentationIconName) and [udisks2 docs](http://udisks.freedesktop.org/docs/latest/udisks.8.html).

### CODE REVIEW

**Comment by mathstuf** [comment]
The branch name is a bit off now, but it's related.

**Comment by coldfix** [comment]
Hey, thanks for your contribution!

I skimmed over your commits. The gist seems fine to me. I noticed two things that should be changed:

- for udisks2, the code should read `self._I.Block.property.XXX` rather than `self._I.XXX`
- rather than patch the `is_handleable` method, just add a rule `IgnoreDevice({'should_automount': False, 'ignore': True})` to the default set of rules (`mount.py`, line 88)

**Comment by mathstuf** [comment]
Updated.

File: udiskie/udisks1.py [hunk 1]
```diff
@@ -275,6 +275,18 @@ def drive(self):
 
     root = drive
 
+    @property
+    def should_automount(self):
+        """Check if the device should be automounted."""
+        return not self._proxy.property.DeviceAutomountHint == 'never'
```
Comment (coldfix) [inline udiskie/udisks1.py]: `return self._proxy.property.DeviceAutomountHint != 'never'`

File: udiskie/mount.py [hunk 2]
```diff
@@ -84,6 +84,7 @@ def __init__(self, udisks,
         self._mount_options = mount_options or (lambda device: None)
         self._ignore_device = ignore_device or FilterMatcher([], False)
         self._ignore_device._filters += [
+            IgnoreDevice({'should_automount': False, 'ignore': True}),
```
Comment (coldfix) [inline udiskie/mount.py]: it'd make more sense to move the newly added ignore rule down by one line:

logically it doesn't make much sense to check for the ignore hint, if the device is not even a block device. But it should work either way.

**Comment by coldfix** [comment]
Okay, looks functional now. I'll do a more rigorous check tomorrow.

I put some minor remarks under the corresponding lines, but don't worry, I can take care of these.

**Comment by coldfix** [comment]
Missing entry `'should_automount'` in the `DeviceFilter.VALID_PARAMETERS` list in `config.py`. Otherwise it tests fine now.

By the way, did you check whether your udisks does provide any icon hints? Mine just returns the empty string for all devices I tested. Therefore udiskie just defaulting to the hardcoded default anyway.

**Comment by mathstuf** [comment]
I set some udev hints for the device to set it and they show up properly :) . Will fix the `VALID_PARAMETERS` list tonight.

**Comment by coldfix** [comment]
Ahh, I see okay, you have some udev rules to make this useful. Nevermind the line in VALID_PARAMETERS, I just added it.

**Pull request merged**